feature/socials

Anastasia-front opened the pull request:
add icons, create Socials component and put it in the header

Anastasia-front committed:
add icons, create Socials component and put it in the header

BogdanGlazkov commented on src/components/Socials/Socials.jsx:
```diff
@@ -0,0 +1,56 @@
+import PropTypes from 'prop-types';
+import Telegram from 'public/icons/telegram.svg';
+import Viber from 'public/icons/viber.svg';
+
```
Відступ між зовнішніми та внутрішніми імпортами

Anastasia-front replied:
виправила

BogdanGlazkov commented on src/components/Socials/Socials.jsx:
```diff
@@ -0,0 +1,56 @@
+};
+
+Socials.propTypes = {
+  component: PropTypes.string,
```
isRequired

Anastasia-front replied:
виправила

BogdanGlazkov commented on src/components/Socials/Socials.jsx:
```diff
@@ -0,0 +1,56 @@
+  const linkClass = {
+    header: 'text-black hover:text-white',
+    burgerMenu: 'text-accent hover:text-white',
+    footer: 'text-bgColor hover:text-accent',
```
Там де hover, додаємо і focus - такий самий

BogdanGlazkov replied:
Ще треба додати плавний перехід, наприклад duration-300

Anastasia-front replied:
виправила

BogdanGlazkov commented on src/components/Socials/Socials.jsx:
```diff
@@ -0,0 +1,56 @@
+          </a>
+        </li>
+      </ul>
+    </>
```
Якщо повертаємо один елемент, додатково огортати його не потрібно

Anastasia-front replied:
так, дякую, забула прибрати фрагмент

nadiakhoptiuk commented on src/components/Socials/Socials.jsx:
```diff
@@ -0,0 +1,56 @@
+            href="#viber"
+            target="_blank"
+            rel="noreferrer noopener nofollow"
+            aria-label="Viber"
```
можна вказати, що це лінк (на обидва)

Anastasia-front replied:
виправила

BogdanGlazkov replied:
Напевно мається на увазі aria-label="Посилання на Viber"

BogdanGlazkov replied:
Зовнішні посилання через тег а, там все правильно було

Anastasia-front replied:
внесла зміни

Anastasia-front committed:
change code according to comments

BogdanGlazkov commented on src/components/Socials/Socials.jsx:
```diff
@@ -0,0 +1,60 @@
+
+import Telegram from 'public/icons/telegram.svg';
+import Viber from 'public/icons/viber.svg';
+
```
Зайві відступи теж не потрібні. Зовнішні імпорти разом (причому PropTypes в кінці, бо використовується в самому низу сторінки), потім відступ, потім внутрішні імпорти

Anastasia-front replied:
зрозуміла, тут виправила, далі всі імпорти буду правильно групувати

BogdanGlazkov commented on src/components/Socials/Socials.jsx:
```diff
@@ -0,0 +1,60 @@
+
+  const linkClass = {
+    header:
+      'text-black hover:text-white focus:text-white transition duration-300',
```
transition тут вказувати не потрібно

BogdanGlazkov commented on src/components/Socials/Socials.jsx:
```diff
@@ -0,0 +1,60 @@
+    header:
+      'text-black hover:text-white focus:text-white transition duration-300',
+    burgerMenu:
+      'text-accent hover:text-white focus:text-white transition duration-300',
```
transition тут вказувати не потрібно

Anastasia-front replied:
видалила

BogdanGlazkov commented on src/components/Socials/Socials.jsx:
```diff
@@ -0,0 +1,60 @@
+    burgerMenu:
+      'text-accent hover:text-white focus:text-white transition duration-300',
+    footer:
+      'text-bgColor hover:text-accent focus:text-accent transition duration-300',
```
transition тут вказувати не потрібно

Anastasia-front replied:
видалила

BogdanGlazkov commented on src/components/Socials/Socials.jsx:
```diff
@@ -0,0 +1,60 @@
+          href="#telegram"
+          target="_blank"
+          rel="noreferrer noopener nofollow"
+          aria-label="Telegram"
```
aria-label="Посилання на Telegram"

Anastasia-front replied:
виправила

Anastasia-front replied:
ще чекаю актуальних посилань на соціальні мережі замовника

Anastasia-front committed:
Merge branch 'dev' into feature/socials

Anastasia-front committed:
change tag Link to a, edit aria-label, delete property transition from linkClass

Anastasia-front committed:
Merge branch 'feature/socials' of https://github.com/SoftRyzen-internship/IM-CONSULTING into feature/socials

nadiakhoptiuk commented on src/components/Socials/Socials.jsx:
```diff
@@ -0,0 +1,55 @@
+  };
+
+  return (
+    <ul className={listClass[component]}>
```
в ідеалі мепнути масив із data (із aria-label, href), додавши іконки.

Anastasia-front committed:
add new social links, put info into data.json and edit styles

import PropTypes from 'prop-types';
import React from 'react';

ці імпорти з бібліотек у файлі Socials.jsx  автоматично сортуються за алфавітом, тому PropTypes вище

nadiakhoptiuk commented on src/components/Socials/Socials.jsx:
```diff
@@ -0,0 +1,73 @@
+};
+
+Socials.propTypes = {
+  component: PropTypes.string.isRequired,
```
тут краще описати масив рядків, які мають приходити, "header" ...

Anastasia-front replied:
виправила

nadiakhoptiuk commented on src/data/home/socials.js:
зроби json і просто перенеси сюди socials масив, а з data прибрати можна

Anastasia-front replied:
зробила

Anastasia-front committed:
moved the array socials from data.json to socials.json and fix propTypes of Socials component

Anastasia-front committed:
Merge branch 'dev' into feature/socials

Anastasia-front committed:
fix imports in Header.jsx
